helios-services/src/main/java/com/spotify/helios/rollingupdate/RollingUpdateOpFactory.java

```diff
@@ -53,13 +58,24 @@ public RollingUpdateOpFactory(final DeploymentGroupTasks tasks,
   }
 
   public RollingUpdateOp start(final DeploymentGroup deploymentGroup,
-                               final RollingUpdateReason reason) {
+                               final RollingUpdateReason reason,
+                               final ZooKeeperClient client) throws KeeperException {
     final List<ZooKeeperOperation> ops = Lists.newArrayList();
     final List<Map<String, Object>> events = Lists.newArrayList();
 
     final List<RolloutTask> rolloutTasks = tasks.getRolloutTasks();
     events.add(eventFactory.rollingUpdateStarted(deploymentGroup, reason));
 
+    final Stat tasksStat = client.exists(
+        Paths.statusDeploymentGroupTasks(deploymentGroup.getName()));
+    if (tasksStat == null) {
+      // Create the tasks path if it doesn't already exist. The following operations (delete or set)
+      // assume the node already exists. If the tasks path is created/deleted before the transaction
+      // is committed it will fail. This will on occasion generate a user-visible error but is
+      // better than having inconsistent state.
+      ops.add(create(Paths.statusDeploymentGroupTasks(deploymentGroup.getName())));
```

Comment:
To me having to pass in the client is real ugly, but I couldn't really figure out a better way to do it. We need to pass it in so that we can do a `exists()` to check if the node exists or not. AFAIK there's no way to do a "create-if-it-doesn't-exist" operation in ZK.